### ISSUE TITLE
Plural keys `key_two`, `key_few`, and `key_many` can't get triggered via `count`

### ISSUE THREAD

**Issue by some1chan** [i18next/i18next#2033]
## 🐛 Bug Report

I'm new to i18next, and I could be missing something _incredibly_ obvious. 

That said, I can't seem to get certain plurals working. I can get `key_zero`, `key_one`, and `key_other` outputs working by changing `count`. Trying to trigger other keys by changing `count` does not work, and only outputs "other".

I'm guessing this is a regression, assuming this feature properly existed somewhere, although I have no clue where (hence not using the regression template). This example works in 23.5.1, and breaks completely on 20.6.1 because of likely API changes. All latest patched major versions afterward (21.10.0 and 22.5.1) exhibit the same behavior.

Translations, keys, and samples copied from [the docs on plurals](https://www.i18next.com/translation-function/plurals#languages-with-multiple-plurals).

## To Reproduce

1. Clone https://github.com/some1chan/i18next-repro
2. Run `npm install`
3. Run `npm run start`

Here's what the `index.js` looks like:
```js
import i18next from "i18next";

await i18next.init({
	lng: "en",
	debug: false,
	resources: {
		en: {
			translation: {
				key_zero: "zero",
				key_one: "singular",
				key_two: "two",
				key_few: "few",
				key_many: "many",
				key_other: "other",
			},
		},
	},
});

console.log(i18next.t("key", { count: 0 }));   // -> "zero"
console.log(i18next.t("key", { count: 1 }));   // -> "singular"
console.log(i18next.t("key", { count: 2 }));   // Should output "two", outputs "other"
console.log(i18next.t("key", { count: 3 }));   // Should output "few", outputs "other"
console.log(i18next.t("key", { count: 4 }));   // Should output "few", outputs "other"
console.log(i18next.t("key", { count: 5 }));   // Should output "few", outputs "other"
console.log(i18next.t("key", { count: 11 }));  // Should output "many", outputs "other"
console.log(i18next.t("key", { count: 99 }));  // Should output "many", outputs "other"
console.log(i18next.t("key", { count: 100 })); // -> "other"

```

## Expected behavior

The console output should output the following, as [outlined in the samples](https://www.i18next.com/translation-function/plurals#languages-with-multiple-plurals):
```js
zero
singular
two
few
few
few
many
many
other
```

## Your Environment

- Node: v18.17.1
- i18next: v23.5.1
- OS: Linux (ARM64)

**Comment by jamuhl** [comment]
rather sure this did not work in 23.5.1 - at least not if setting lng to en:

- pluralrules for en are https://www.unicode.org/cldr/cldr-aux/charts/36.1/supplemental/language_plural_rules.html#en so there is only the select for one and other (plus the zero we added for convinience)

**Comment by some1chan** [comment]
Sounds about right. A friend just told me that `two` doesn't exist in plural rules, and linked me to [Intl.PluralRules() constructor - JavaScript | MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules/PluralRules), with the following:

![image](https://github.com/i18next/i18next/assets/31833556/4cb0d462-b9ba-460f-bd5f-5a33c18dec3e)

This probably just means this just a documentation issue, specifically for clarity. I interpreted on the site "Sample uses arabic which has 5 plural forms beside the singular" to be "This sample uses Arabic numerals, which has five plural forms beside the singular." The text in there being solely English was also confusing, as it wasn't Arabic (the language) despite it claiming to be Arabic. I presumed numerals from there had a specific system, and outside it was different.

Perhaps a note like this would help:
> The language you're localizing to may have different plural rules, and you should check with this [JSFiddle utility](https://jsfiddle.net/6bpxsgd4). Although Arabic has `_zero`, `_one`, `_two`, `_few`, `_many`, and `_other`, English only has `_zero`, `_one`, and _`other`.
